pkg/controllers/state/suite_test.go

```diff
@@ -64,17 +67,55 @@ var _ = AfterSuite(func() {
 })
 
 var _ = BeforeEach(func() {
+	// fmt.Printf("Running setup...\n")
 	cloudProvider = &fake.CloudProvider{InstanceTypes: fake.InstanceTypesAssorted()}
 	cluster = state.NewCluster(env.Client, cloudProvider)
+	provisioner = test.Provisioner(test.ProvisionerOptions{ObjectMeta: metav1.ObjectMeta{Name: "default"}})
 	nodeController = state.NewNodeController(env.Client, cluster)
 	podController = state.NewPodController(env.Client, cluster)
-	provisioner = test.Provisioner(test.ProvisionerOptions{ObjectMeta: metav1.ObjectMeta{Name: "default"}})
+	metricScraper = statemetrics.NewMetricScraper(ctx, cluster)
 	ExpectApplied(ctx, env.Client, provisioner)
 })
+
 var _ = AfterEach(func() {
+	metricScraper.Terminate()
 	ExpectCleanedUp(ctx, env.Client)
 })
 
+var _ = Describe("Node Metrics", func() {
+	It("should update the allocatable metric", func() {
+		node := test.Node(test.NodeOptions{
+			Allocatable: v1.ResourceList{
+				v1.ResourcePods:   resource.MustParse("100"),
+				v1.ResourceCPU:    resource.MustParse("5000"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+		})
+		ExpectApplied(ctx, env.Client, node)
+		ExpectReconcileSucceeded(ctx, nodeController, client.ObjectKeyFromObject(node))
+
+		// metrics should now be tracking the allocatable capacity of our single node
+		metricScraper.Update()
+		nodeAllocation := ExpectMetric("karpenter_nodes_allocatable")
+
+		expectedValues := map[string]float64{
+			"cpu":    5000.0,
+			"pods":   100.0,
+			"memory": 32 * 1024 * 1024 * 1024,
+		}
```

Comment:
Can we just copy the resource list and use the resource functions to convert before checking?